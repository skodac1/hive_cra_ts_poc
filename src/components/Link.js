import React from 'react'
import { Text } from 'rebass/styled-components'
import PropTypes from 'prop-types'

const Link = ({ text, url }) => {
  return (<Text
    as="a"
    href={url}
    mx={2}
    fontWeight="body"
    fontSize={2}
    sx={{
      cursor: 'pointer',
      textDecoration: 'none',
      color: 'white',
    }}
  >
    {text}
  </Text>)
}

Link.propTypes = {
  text: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
}
export default Link
