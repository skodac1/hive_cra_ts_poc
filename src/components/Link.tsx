import React from 'react'
import { Text } from 'rebass/styled-components'

type LinkProps = {
    text: string,
    url: string
}

const Link = ({ text, url }: LinkProps) => {
  return (<Text
    as="a"
    href={url}
    mx={2}
    fontWeight="body"
    fontSize={2}
    sx={{
      cursor: 'pointer',
      textDecoration: 'none',
      color: 'white',
    }}
  >
    {text}
  </Text>)
}

export default Link
