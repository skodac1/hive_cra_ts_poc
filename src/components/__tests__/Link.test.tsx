import * as React from 'react';
import { render } from '@testing-library/react';
import Link from "../Link";

describe('<Sidebar />', () => {
    it('matches snapshot', () => {
        const { baseElement } = render(<Link text="text" url="url" />);
        expect(baseElement).toMatchSnapshot();
    });
})