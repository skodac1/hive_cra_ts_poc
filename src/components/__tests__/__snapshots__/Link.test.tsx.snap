// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Sidebar /> matches snapshot 1`] = `
<body>
  <div>
    <a
      class="sc-bczRLJ kiajAT"
      font-size="2"
      font-weight="body"
      href="url"
    >
      text
    </a>
  </div>
</body>
`;
